GAME_DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,122 @@
+
+### Explain when Needed
+If something progress-related has no obvious explanation, the player will have to rely on a wiki to find out how to progress. One example is the bellows; players won't know to place it next to a furnace and jump on it repeatedly. The issue can be fixed by adding a tooltip that tells them how it works, while leaving out details they can intuitively find out themselves.
+
+Having a mechanic that isn't explained or illuded to makes it an "Invisible Mechanic." These can only be found out through looking at the game's code, wiki reading, or asking a developer, thus giving a disadvantage to casual players. It's best to avoid these at all costs.
+
+### Micro Moments
+These are tiny things the player does in between larger events; examples include:
+- Travelling
+- Parkour
+- Mining
+- Building
+- Crafting
+- Managing inventory
+- Fighting
+
+These moments are extremely important as they let the player mentally rest, so making sure they're as satisfying and consistent as possible is a must.
+
+### Make the World Feel Alive
+To add immersion to the game, creatures should perform behaviors outside of player input, such as hunting, migrating, eating, or sleeping.
+
+# Balancing
+When balancing the game, keep in mind how players might interact with the world, the wildlife, and each other.
+
+### 2OP4ME
+At no point should the player be extremely hard to kill. Armor, tools, accessories, and buffs should be designed around aiding the player in skill-based encounters, not making them win regardless of skill.
+
+### Trade-offs
+If the player is given something to aid them, then it should have an appropriate take-away to balance it.
+
+**Examples:**
+- Using rare resources to create a strong tool.
+- An accessory that heals the player, but takes away energy or some other resource.
+- Enemies have strengths and weaknesses towards particular damage types
+
+### Mob Greifing
+Creatures should not be able to grief players without their input. Destructive creatures are reserved for caves.
+
+# Little Details
+The following contains spoilers for future content. If you are not looking into contributing, you should not read this section.
+<details>
+  <summary>Open Spoilers</summary>
+
+### Big Trees vs Small Trees
+There are two categories of trees, big and small. Big trees are designed to be built upon or left as decoration, whereas small trees are designed to be chopped down.
+
+### World Building with Gameplay
+Gameplay elements have implications on the world's lore, and the same goes other way around. It's best to design the gameplay element first, and then apply the world's style to the element during implementation.
```

Comment:
I think we have to be careful here, if we always go for gameplay without thinking about world building, then we will end up with a mess of fun gameplay features and no way to build a cohesive world around that.